tdms/src/argument_parser.cpp

```diff
@@ -0,0 +1,120 @@
+#include "stdexcept"
+#include "argument_parser.h"
+
+using namespace std;
+
+
+ArgumentParser::ArgumentParser() = default;
+
+ArgumentNamespace ArgumentParser::parse_args(int nargs, char *argv[]) {
+
+  auto args = ArgumentNamespace(nargs, argv);
+
+  if (args.have_flag("-h")){
+    print_help_message();
+    exit(0);
+  }
+
+  if( (args.num_non_flag != 2) &&  (args.num_non_flag != 3) ){
+    fprintf(stderr,"incorrect number of arguments\n");
+    exit(-1);
+  }
+
+  if( args.num_non_flag == 2){
+    fprintf(stdout,"infile:[%s], outfile:[%s], m=%d\n",
+            args.input_filename(),
+            args.output_filename(),
+            args.have_flag("-m"));
+  }
+  else{
+    fprintf(stdout,"infile:[%s], gridfile:[%s], outfile:[%s], m=%d\n",
+            args.input_filename(),
+            args.grid_filename(),
+            args.output_filename(),
+            args.have_flag("-m"));
+  }
+
+  return args;
+}
+
+void ArgumentParser::print_help_message(){
+  fprintf(stdout,"Usage:\n"
+                 "openandorder [options] infile outfile\n"
+                 "openandorder [options] infile gridfile outfile\n"
+                 "Options:\n"
+                 "-h:\tDisplay this help message\n"
+                 "-m:\tMinimise output file size by not saving vertex and facet information\n\n");
+}
+
+
+ArgumentNamespace::ArgumentNamespace(int nargs, char *argv[]) {
+
+  arguments = std::vector<std::string>(argv + 1, argv + nargs);
+
+  for (const auto& arg: arguments){
+
+    if (!is_a_flag_argument(arg)){
+      non_flag_arguments.push_back(arg);
+      num_non_flag++;
+    }
+  }
+
+}
+
+bool ArgumentNamespace::is_a_flag_argument(std::string arg){
+  return arg[0] == '-';
+}
+
+bool ArgumentNamespace::have_flag(std::string const &flag){
+
+  for (const auto& arg : arguments){
+    if(arg == flag) return true;
+  }
+
+  return false;
+}
+
+const char* ArgumentNamespace::output_filename() {
+
+  if (num_non_flag == 3){
+    return non_flag_arguments[2].c_str();
+  } else if (num_non_flag == 2){
+    return non_flag_arguments[1].c_str();
+  }
+
+  throw std::runtime_error("Failed to determine the output file from arguments");
+}
+
+const char* ArgumentNamespace::grid_filename() {
+
+  if (num_non_flag == 3){
```

Comment:
```suggestion
  if has_grid_filename(){
```
Might as well use the method to make intent clear?

Reply:
Definitely (not going to commit your suggestion because it's missing some brackets 😅)